remotes/docker/authorizer.go

```diff
@@ -276,8 +278,7 @@ func (ah *authHandler) doBearerAuth(ctx context.Context) (token string, err erro
 			err = errors.Wrap(err, "failed to fetch oauth token")
 		}()
 		// credential information is provided, use oauth POST endpoint
-		// TODO: Allow setting client_id
-		resp, err := auth.FetchTokenWithOAuth(ctx, ah.client, ah.header, "containerd-client", to)
+		resp, err := auth.FetchTokenWithOAuth(ctx, ah.client, ah.header, containerdClientID, to)
```

Comment:
`containerdClientID` remains constant, so this PR seems NOP

Reply:
thought it would be the same way as https://github.com/containerd/containerd/blob/main/runtime/v2/shim.go#L47
which could change when build with `-ldflags`

Reply:
No, you can't do that for `const`.
Did you really test your PR?

Reply:
And why do you need to change the value?

Reply:
Not yet, just find there is a TODO when reading the code and try to fix it.

Reply:
My bad, `-ldflags` only works for variables not const. Can u tell me the correct way to change https://github.com/containerd/containerd/blob/main/runtime/v2/shim.go#L47? by timeout.Set?

Reply:
To fix this TODO comment, do i need to add a filed to `authHandler` ?